sdk/storage/azure-storage-common/src/main/java/com/azure/storage/common/Utility.java

```diff
@@ -86,11 +86,11 @@
      * @param queryString Query string to parse
      * @return a mapping of query string pieces as key-value pairs.
      */
-    public static TreeMap<String, String[]> parseQueryStringSplitValues(final String queryString) {
+    public static Map<String, String[]> parseQueryStringSplitValues(final String queryString) {
         return parseQueryStringHelper(queryString, (value) -> urlDecode(value).split(","));
     }
 
-    private static <T> TreeMap<String, T> parseQueryStringHelper(final String queryString,
+    private static <T> Map<String, T> parseQueryStringHelper(final String queryString,
```

Comment:
reg line 102: just double checking, is it always required that query string part of SaS url to be in `a=b` format? not sure about rules around blob url, any chance it could be something like `a=b&c=&d&e=f` here c and d has no value.

Reply:
Discussed offline, there is an expectation for the query parameters to have values. This is a small guard that should be done anyways in the future.